sway/commands/opacity.c

```diff
@@ -1,21 +1,48 @@
 #include <assert.h>
 #include <stdlib.h>
+#include <strings.h>
 #include "sway/commands.h"
 #include "sway/tree/view.h"
 #include "log.h"
 
-static bool parse_opacity(const char *opacity, float *val) {
+static bool adjust_opacity(int argc, char * const argv[], float *alpha) {
 	char *err;
-	*val = strtof(opacity, &err);
-	if (*val < 0 || *val > 1 || *err) {
+	float val;
+
+	if (argc == 1) {
```

Comment:
I'm not a huge fan of the repetition in this function. Something like the following would be better:
```c
char *err;
float val = strtof(argc == 1 ? argv[0] : argv[1], &err);
if (*err) {
    return false;
}

if (strcasecmp(argv[0], "plus") == 0) {
    val = *alpha + val;
} else if (strcasecmp(argv[0], "minus") == 0) {
    val = *alpha - val;
} else if (argc > 1 && strcasecmp(argv[0], "set") != 0) {
    return false;
}

if (val < 0 || val > 1) {
    return false;
}

*alpha = val;
return true;
```

Reply:
Nearly copied this exactly.